packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -0,0 +1,386 @@
+              'By default, all build configurations are built.'
+      )
+      ..addFlag('universal',
+        help: 'Produce universal frameworks that include all valid architectures. '
+              'This is true by default.',
+        defaultsTo: true,
+        negatable: true
+      )
+      ..addFlag('xcframework',
+        help: 'Produce xcframeworks that include all valid architectures (Xcode 11 or later).',
+      )
+      ..addOption('output',
+        abbr: 'o',
+        valueHelp: 'path/to/directory/',
+        defaultsTo: fs.path.join(fs.currentDirectory.path, 'build', 'ios', 'framework'),
+        help: 'Location to write the frameworks.',
+      );
+  }
+
+  AotBuilder aotBuilder;
+  BundleBuilder bundleBuilder;
+
+  @override
+  final String name = 'ios-framework';
+
+  @override
+  final String description = 'Produces a .framework directory for a Flutter module '
+      'and its plugins for integration into existing, plain Xcode projects.\n'
+      'This can only be run on macOS hosts.';
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{
+    DevelopmentArtifact.universal,
+    DevelopmentArtifact.iOS,
+  };
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    final FlutterProject flutterProject = FlutterProject.current();
+    if (!flutterProject.isModule) {
+      throwToolExit('Building frameworks for iOS is only supported from a module.');
+    }
+
+    if (getCurrentHostPlatform() != HostPlatform.darwin_x64) {
+      throwToolExit('Building frameworks for iOS is only supported on the Mac.');
+    }
+
+    if (argResults['xcframework'] && xcode.majorVersion < 11) {
+      throwToolExit('--xcframework requires Xcode 11.');
+    }
+
+    if (!argResults['universal'] && !argResults['xcframework']) {
+      throwToolExit('--universal or --xcframework is required.');
+    }
+    final List<BuildMode> buildModes = <BuildMode>[];
+
+    if (argResults['debug']) {
+      buildModes.add(BuildMode.debug);
+    }
+
+    if (argResults['profile']) {
+      buildModes.add(BuildMode.profile);
+    }
+
+    if (argResults['release']) {
+      buildModes.add(BuildMode.release);
+    }
+
+    if (buildModes.isEmpty) {
+      throwToolExit('--debug or --profile or --release is required.');
+    }
+
+    final String outputArgument = argResults['output'];
+
+    if (outputArgument.isEmpty) {
+      throwToolExit('--output is required.');
+    }
+
+    final BuildableIOSApp iosProject = await applicationPackages.getPackageForPlatform(TargetPlatform.ios);
+
+    if (iosProject == null) {
+      throwToolExit("Module's iOS folder missing");
+    }
+
+    final Directory outputDirectory = fs.directory(fs.path.normalize(outputArgument));
+
+    if (outputDirectory.existsSync()) {
+      outputDirectory.deleteSync(recursive: true);
+    }
+
+    aotBuilder ??= AotBuilder();
+    bundleBuilder ??= BundleBuilder();
+
+    for (BuildMode mode in buildModes) {
```

Comment:
You could speed this up a bit by running these concurrently. That might not be worth doing for now, since you'd get it for free via assemble